src/kfactory/pin.py

```diff
@@ -0,0 +1,367 @@
+        index.
+        """
+        return self.kcl.find_layer(self.layer_info, allow_undefined_layers=True)
+
+    @property
+    def layer_info(self) -> kdb.LayerInfo:
+        """Get the layer info of the port.
+
+        This corresponds to the port's cross section's main layer.
+        """
+        return self.base.layer_info
+
+    def __eq__(self, other: object) -> bool:
+        """Support for `pin1 == pin2` comparisons."""
+        if isinstance(other, ProtoPin):
+            return self._base == other._base
+        return False
+
+    @property
+    @abstractmethod
+    def pos(self) -> PointLike[TUnit]: ...
+
+    @pos.setter
+    @abstractmethod
+    def pos(self, value: PointLike[TUnit]) -> None: ...
+
+    def to_itype(self) -> Pin:
+        """Convert the port to a dbu port."""
+        return Pin(base=self._base)
+
+    def to_dtype(self) -> DPin:
+        """Convert the port to a um port."""
+        return DPin(base=self._base)
+
+    @property
+    def allowed_angles(self) -> list[Literal[0, 1, 2, 3]]:
+        """Angle of the transformation.
+
+        In the range of `[0,1,2,3]` which are increments in 90°.
+        """
+        return self._allowed_angles
+
+    @allowed_angles.setter
+    def allowed_angles(self, value: list[Literal[0, 1, 2, 3]]) -> None:
+        self._allowed_angles = value
+
+    @property
+    def allowed_orientations(self) -> list[Literal[0, 90, 180, 270]]:
+        """Returns orientation in degrees for gdsfactory compatibility.
+
+        In the range of `[0,360)`
+        """
+        return [a * 90 for a in self.base.allowed_angles]  # type: ignore[misc]
+
+    @allowed_orientations.setter
+    def allowed_orientations(self, value: Sequence[float]) -> None:
+        """Set the orientation of the port."""
+        values = {(a % 360) / 90 for a in value}
+        if values - {0, 1, 2, 3} is not None:
+            raise ValueError("Only {0,90,180,270} are allowed as orientation.")
+        self.base.allowed_angles |= values
+
+    @abstractmethod
+    def copy(
+        self,
+        trans: kdb.Trans | kdb.DCplxTrans = kdb.Trans.R0,
+        post_trans: kdb.Trans | kdb.DCplxTrans = kdb.Trans.R0,
+    ) -> ProtoPin[TUnit]:
+        """Copy the port with a transformation."""
+        ...
+
+    @property
+    def center(self) -> tuple[TUnit, TUnit]:
+        """Returns port center."""
+        return (self.pos.x, self.pos.y)
+
+    @center.setter
+    def center(self, value: tuple[TUnit, TUnit]) -> None:
+        self.x = value[0]
+        self.y = value[1]
+
+    @property
+    def x(self) -> TUnit:
+        """X coordinate of the port."""
+        return self.pos.x
+
+    @x.setter
+    @abstractmethod
+    def x(self, value: TUnit) -> None: ...
+
+    @property
+    def y(self) -> TUnit:
+        """Y coordinate of the port."""
+        return self.pos.y
+
+    @y.setter
+    @abstractmethod
+    def y(self, value: TUnit) -> None: ...
```

Comment:
```suggestion
    @y.setter
    def y(self, value: TUnit) -> None:
        self.pos = (self.x, value)
```
I think we should be able to do this?

Reply:
Can yes, but imo `pos` should be like `trans`/`dcplx_trans` in Port. If you want tuple, use `center`